File: 227/227.cpp
#include<iostream>

using namespace std;

// Failing with Time Limits 250 ms
int main()
{
	long library_books; cin >> library_books;
	long my_books; cin >> my_books;
	long week_day; cin >> week_day;

	long count = 0;
	long reads = 1;
	
	if ((week_day == 6 || week_day == 7) && (my_books == 0)){
		cout << 0;
		return 0;
	}

	if (week_day == 6 && (my_books == 1 || my_books == 2)) {
		cout << 1;
		return 0;
	}

	while (true) {
		if (week_day < 6) {
			my_books += library_books;
		}
		my_books = my_books - reads;

		if (my_books < 0) {
			break;
		}

		week_day = (week_day % 7) + 1;
		count++;
		reads++;
	}

	cout << count;
	return 0;
}

File: 227/227.go
package main

import "fmt"
// (Just for Fun)
// Failing with Time Limits 250 ms (Just for Fun)
func main() {
    var library_books int64
    fmt.Scan(&library_books)
    var my_books int64
    fmt.Scan(&my_books)
    var week_day int64
    fmt.Scan(&week_day)
    count := 0
    reads := int64(1)
    for {
        if week_day < 6 {
            my_books += library_books
        }
        my_books -= reads
        if my_books < 0 {
            break
        }
        week_day = (week_day % 7) + 1
        count++
        reads++
    }
    fmt.Println(count)
}

